cms/djangoapps/contentstore/course_info_model.py

```diff
@@ -60,17 +60,21 @@ def update_course_updates(location, update, passed_id=None, user=None):
 
     if passed_id is not None:
         passed_index = _get_index(passed_id)
-        # oldest update at start of list
-        if 0 < passed_index <= len(course_update_items):
-            course_update_dict = course_update_items[passed_index - 1]
-            course_update_dict["date"] = update["date"]
-            course_update_dict["content"] = update["content"]
-            course_update_items[passed_index - 1] = course_update_dict
-        else:
+
+        # if passed_index in course_update_items_ids:
+        for course_update_item in course_update_items:
+            if course_update_item["id"] == passed_index:
+                course_update_dict = course_update_item
+                course_update_item["date"] = update["date"]
+                course_update_item["content"] = update["content"]
+        if course_update_dict is None:
             return HttpResponseBadRequest(_("Invalid course update id."))
     else:
+        course_update_items_ids = [course_update_item['id'] for course_update_item in course_update_items]
```

Comment:
As we don't delete the items from backend, I guess `max(course_update_items_ids) + 1 ` would be same as `len(course_update_items) + 1`?

Reply:
This `len(course_update_items) + 1` was actually causing the problem on the prod as it can make 2 course_updates having the same id.

Reply:
got it.